### ISSUE TITLE
Graph/tree automation broken in 1.1.2

### ISSUE THREAD

**Issue by stormonts** [Cacti/cacti#520]
In version 1.1.1, I had graph and tree rules created to create graphs and then add them to the tree.  Now in version 1.1.2, none of the rules are running (either automatically or via the "Apply Automation rules" option.  I don't see any errors in the logs, the graphs just aren't created or added to the tree.  In addition, the graph rules always show "no matching devices", even though I know I have devices that match the rules (and had matched the rules in 1.1.1)

**Comment by dylan-clear** [comment]
I encountered this on a clean install of 1.1.2. The logs showed SQL queries failing as per:

```
2017-04-05 18:58:05 - CMDPHP SQL Backtrace: (/automation_graph_rules.php: 74 automation_graph_rules_edit)(/automation_graph_rules.php: 600 display_matching_hosts)(/lib/api_automation.php: 260 db_fetch_assoc)(/lib/database.php: 320 db_fetch_assoc_prepared)(/lib/database.php: 359 cacti_debug_backtrace)
2017-04-05 18:58:05 - DBCALL ERROR: SQL Assoc Failed!, Error: Unknown column 'h.hostname' in 'where clause'
2017-04-05 18:58:05 - DBCALL ERROR: SQL Assoc Failed!, Error:1054, SQL:"SELECT h.id AS host_id, h.hostname, h.description, h.disabled, h.status, ht.name AS host_template_name FROM host AS h LEFT JOIN host_template AS ht ON (h.host_template_id=ht.id) WHERE ( `h.hostname` IS NOT NULL AND `h.snmp_version` > '1') ORDER BY description ASC LIMIT 0,30"
```
It looks like someone inserted backticks where there should have been single quotes...

I made the following changes to the file /opt/cacti/lib/api_automation.php:

Line 1490 becomes:
`$sql_having .= ($i == 0 ? '':' OR ') . '\'' . $column['field_name'] . '\'' . ' LIKE "%' . $filter . '%"';`

Line 1595 becomes:
`$sql_filter .= (' ' . $prefix . '\'' . $automation_rule_item['field'] . '\'');`

Basically just swap any ` characters for \'

Solved the problem for me.

**Comment by reboot1983** [comment]
This doesn't seems right for me. The backticks are not at the right place, but should not be single quotes.
it should be: \`h\`.\`hostname\` instead of 'h.hostname' or `h.hostname`.  'h.hostname' makes it just a string comparison (h.hostname just became a string, and not a table.column reference)

**Comment by dylan-clear** [comment]
You are correct, that actually only worked for the single query I was running :-) I make no claim as to understanding the code tbh. I just tested the SQL query in the logs on the CLI, it didn't work, then I changed the backtickls to single quotes and it did work. So I swapped the backticks in the code for single quotes... And it seemed to work. Sorry.

**Comment by reboot1983** [comment]
@cigamit added a patch for this. Please revise :)

[Issue_520.patch.zip](https://github.com/Cacti/cacti/files/900132/Issue_520.patch.zip)

**Comment by dylan-clear** [comment]
The patch doesn't solve the problem for me - backticks still appear in the SQL query and SQL doesn't like them. I think maybe that patch solved the issue around line 1479, but not for the backticks in line 1595?

```
2017-04-06 09:57:02 - CMDPHP SQL Backtrace: (/automation_graph_rules.php: 74 automation_graph_rules_edit)(/automation_graph_rules.php: 600 display_matching_hosts)(/lib/api_automation.php: 260 db_fetch_assoc)(/lib/database.php: 320 db_fetch_assoc_prepared)(/lib/database.php: 359 cacti_debug_backtrace)
2017-04-06 09:57:02 - DBCALL ERROR: SQL Assoc Failed!, Error: Unknown column 'h.hostname' in 'where clause'
2017-04-06 09:57:02 - DBCALL ERROR: SQL Assoc Failed!, Error:1054, SQL:"SELECT h.id AS host_id, h.hostname, h.description, h.disabled, h.status, ht.name AS host_template_name FROM host AS h LEFT JOIN host_template AS ht ON (h.host_template_id=ht.id) WHERE ( `h.hostname` LIKE '%rcbc%' AND `h.snmp_version` > '1') ORDER BY description ASC LIMIT 0,30"
```

**Comment by cigamit** [comment]
Resolved.  Sorry about that...
